include/swift/AST/TypeRepr.h

```diff
@@ -613,38 +613,38 @@ class NamedTypeRepr : public TypeRepr {
 
 /// \brief A protocol composite type.
 /// \code
-///   protocol<Foo, Bar>
+///   Foo & Bar
 /// \endcode
 class ProtocolCompositionTypeRepr : public TypeRepr {
   ArrayRef<IdentTypeRepr *> Protocols;
-  SourceLoc ProtocolLoc;
-  SourceRange AngleBrackets;
+  SourceLoc FirstTypeLoc;
+  SourceRange CompositionRange;
```

Comment:
Ah, in particular I think it makes sense for diagnostics to point to the `&` rather than the first type, like they do for binary operator expressions.